src/python/dxpy/bindings/dxdataobject_functions.py

```diff
@@ -63,7 +63,17 @@ def is_dxlink(x):
     Returns whether *x* appears to be a DNAnexus link (is a dict with
     key ``"$dnanexus_link"``) with a referenced data object.
     '''
-    return isinstance(x, dict) and '$dnanexus_link' in x and (isinstance(x['$dnanexus_link'], basestring) or isinstance(x['$dnanexus_link'], dict) and 'id' in x['$dnanexus_link'])
+    if not isinstance(x, dict):
+        return False
+    if '$dnanexus_link' not in x:
+        return False
+    link = x['$dnanexus_link']
+    if isinstance(link, basestring):
+        return True
```

Comment:
The reason I unwrapped the conditionals here and above is that the line was far too long and hard to read (including relying on multiple operator precedence rules without parentheses), and dereferenced the dictionary key multiple times. The logic remains the same.